pkg/querier/querier.go

```diff
@@ -60,7 +59,6 @@ type Config struct {
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	cfg.Engine.RegisterFlagsWithPrefix("querier", f)
 	f.DurationVar(&cfg.TailMaxDuration, "querier.tail-max-duration", 1*time.Hour, "Limit the duration for which live tailing request would be served")
-	f.DurationVar(&cfg.QueryTimeout, "querier.query-timeout", 1*time.Minute, "Timeout when querying backends (ingesters or storage) during the execution of a query request")
```

Comment:
We should aim to deprecate options first, then remove.
A missing flag will cause the application to not start AFAIK.
All flag changes should be accompanied by a changelog/upgrade guide entry

Reply:
I added a deprecated notice; WDYT?

Reply:
Nice, thanks!